server/channels/app/brand.go

```diff
@@ -45,7 +46,27 @@ func (a *App) SaveBrandImage(rctx request.CTX, imageData *multipart.FileHeader)
 	}
 
 	t := time.Now()
-	a.MoveFile(BrandFilePath+BrandFileName, BrandFilePath+t.Format("2006-01-02T15:04:05")+".png")
+	// Try to backup the old brand image if it exists
+	oldPath := BrandFilePath + BrandFileName
+	newPath := BrandFilePath + t.Format("2006-01-02T15:04:05") + ".png"
+
+	fileExists, fileErr := a.FileExists(oldPath)
+	if fileErr != nil {
+		rctx.Logger().Warn("Failed to check if brand image exists before backup", mlog.String("path", oldPath), mlog.Err(fileErr))
+	}
+
```

Comment:
Shouldn't `a.MoveFile` already return an error if oldPath doesn't exist? Do we need this additional call? Another problem with this is that fileExists + moveFile is not atomic. FileExists might return true, and then before calling MoveFile, something else might have removed the file.

Reply:
I'm less concerned about the atomic nature. If the old file gets removed while the new one tries to replace it, we just log an error.

Reply:
That's true, but won't it have the same error from within MoveFile itself if the oldPath doesn't exist?

Reply:
Correct. `MoveFile` would return an error, which we would log. First checking with `a.FileExists` avoids logging an error.

Reply:
Ah okay, if avoiding logging an error if file is not found is our objective, we can simply inspect the error returned and if it's for file not found, then change to warn instead of error. I believe both localStore and s3store should return unique errors for that.

I don't want to unnecessarily hold up this PR, but it just feels like a roundabout solution to me, especially when it anyways logs the error because it's not atomic.

If we really want to avoid logging the error, which I don't think is that big of a deal. Then we should simply inspect the error returned and change to warn instead of error.

LMK what you think.

Reply:
I've spent some time looking into the different implementations of `MoveFile` to find a dedicated error to check, but I couldn't find one. I'm strongly leaning towards not adding such a custom check.

Either we continue with the code as it is or we just drop the error check for `MoveFile.

Reply:
Ok, I checked the implementations and it looks like we simply override the file path if it exists. So actually there is never an error thrown. Checking for the file before moving actually changes the logic slightly.

Since the filename is granular down to the second, this will only collide if two requests happen at the same second. 

I am thinking that checking if the file exists or not is a bit redundant given the logic. I do agree that atomicity does not play a big role here because the file name is scoped to a second. But since collisions will anyways be rare, checking for the file name doesn't hurt as well.